### DISCUSSION TITLE
Can I run github actions on my laptop?

### DISCUSSION THREAD

**Discussion by sekka1** [community/community#25711] [Actions]
<p>Is there a way to run the github action on my laptop?</p>
<p>Instead of having to commit and push to test out the workflows, is there a way I can test it out locally?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>After you create an action, It could not run directly in your local laptop. You need to test it out in workflows.  <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fcreating-a-javascript-action%23testing-out-your-action-in-a-workflow&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Cdfaf041e0e35470402e508d7717311ee%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637102611342198646&amp;sdata=HDAN7ZcAScCnK1x0kUdYKDZZ0Zyf1rgqeHxNfwhqLzE%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-a-javascript-action#testing-out-your-action-in-a-workflow</a></p>
<p>And a workflow need to has a <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fevents-that-trigger-workflows&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Cdfaf041e0e35470402e508d7717311ee%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637102611342198646&amp;sdata=bUaMYKhjGSVHkOlhox%2FVf0n%2FQzX5lIZkZxKxEiDzfiA%3D&amp;reserved=0" rel="nofollow noopener">trigger event</a>, on: push is an easier way.</p>
<p>There are some Development tools for GitHub Actions, they can help you create actions more quickly: <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fdevelopment-tools-for-github-actions&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Cdfaf041e0e35470402e508d7717311ee%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637102611342208639&amp;sdata=PYE0B8fcsF4RNtJt6mHacSsEVafoVmZMP7%2F%2FOxvEzjk%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions</a></p>

**Comment by iwarner** [comment]
<p>The solutions does not answer the question:</p>
<p>There is a tool to locally test actions:</p>
<p><a href="https://github.com/nektos/act" rel="nofollow noopener">https://github.com/nektos/act</a></p>
<p>Try this out.</p>

**Comment by cardinalby** [comment]
<p>If you want to test/debug an individual action (not the entire workflow) you can use <a href="https://github.com/cardinalby/github-action-ts-run-api" rel="noopener nofollow ugc">github-action-ts-run-api</a> JS package. But it’s mainly for actions authors who want to write integration tests for their actions and run them both locally and on CI.</p>
